### ISSUE TITLE
sodium_init() fails because already initialized

### ISSUE THREAD

**Issue by asn-the-goblin-slayer** [pyca/pynacl#186]
Hello,

I'm using PyNaCl in a new secure messaging project: https://github.com/asn-the-goblin-slayer/viola

I'm experiencing problems with `sodium_core.py:sodium_init()` failing to initialize sodium, because apparently it has already been initialized. Specifically it hits the segment below:

```
def sodium_init():
    """
    Initializes sodium, picking the best implementations available for this
    machine.
    """
    retval = lib.sodium_init()
    if retval != 0:
        raise CryptoError("Could not initialize sodium %d" % retval)
```

and `lib.sodium_init()` returns `1` because it has already been initialized somehow. I think this might have to do
with the way I import nacl, and the way pynacl initializes libsodium without checking if it has already been inited.

Unfortunately, while I could reproduce this issue in the past, I can't anymore. But I just received a bug report about it:
https://github.com/asn-the-goblin-slayer/viola/issues/2

Wouldn't it be more appropriate to not call `sodium_init()` if it has already been initialized? Or alternatively, to not raise an exception if `lib.sodium_init()` returns 1 (already initialized)?

Or are there any guidelines on how to import nacl modules so that this doesn't happen? 
For example here is how I import nacl submodules in my project:

```
./viola/crypto.py:8:import nacl.signing
./viola/crypto.py:9:import nacl.encoding
./viola/crypto.py:10:import nacl.secret
./viola/crypto.py:11:import nacl.utils
./viola/viola.py:16:import nacl.exceptions # XXX dirty
./viola/room.py:3:import nacl.utils # XXX dirty. only crypto should touch nacl. move there!!!
./viola/introduction.py:10:import nacl.exceptions # XXX dirty
```

This issue has been quite annoying while deploying viola to other systems :(
It would be great if it could be resolved somehow!

Thanks!

**Comment by asn-the-goblin-slayer** [comment]
Hello, I also provided a patch here: 
https://github.com/pyca/pynacl/pull/187

**Comment by lmctv** [comment]
@asn-the-goblin-slayer I think this could be closed, if #199 works for you too.

**Comment by asn-the-goblin-slayer** [comment]
Yep, closing this. Thanks!
